BDMS-221-225: Core well information model updates

ksmuczynski opened the pull request:
<!--
- BDMS-221-225: Core well information model updates
-->
### **Why**
_This PR addresses the following problem / context:_
- This PR addresses updates to the Group model. It also addressed updates to the lexicon related to the `Group` and `StatusHIstory` models.
- The purpose of this PR is to make these intermediate updates available to @jacob-a-brown as we continue to co-work on the main BDMS-221 task.
### **How**
_Implementation summary - the following was changed / added / removed:_
- Refactored Group model in db/group.py, adding the `group_type` and `monitoring_frequency` fields.
- Added new categories and terms to core/lexicon.json to support expanded group types and monitoring frequencies.
- Renamed `well_status` category to `status_value`.
- Added new `status_type` category and related terms.
- Updated `status_type` term definitions.
### **Notes**
_Any special considerations, workarounds, or follow-up work to note?_
- This PR does NOT address updates to the schemas, tests, or transfer scripts.

ksmuczynski committed:
feat: update group model and expand lexicon

Refactored Group model in db/group.py, adding the `group_type` and `monitoring_frequency` fields.

Added new categories and terms to core/lexicon.json to support expanded group types and monitoring frequencies.

ksmuczynski committed:
Merge branch 'bdms-221' into kas-bdms-221-225-core-well-info-models-schemas

ksmuczynski committed:
refactor: update lexicon category names and terms associated with the StatusHistory table.

`well_status` is defined as a lexicon category, but it should be a lexicon value associated with the `status_type` category.

 The `status_type` category does not exist, but it should.

The terms assigned to the `well_status` category should actually be assigned to the `status_value` category.

Renamed `well_status` category to `status_value`.
Added new `status_type` category and related terms.
Updated term definitions.

ksmuczynski committed:
refactor: list status_type terms in proper case.

ksmuczynski committed:
refactor: update lexicon

`Monitoring Status` should be a value/term associated with the `status_type` category, not a category unto itself.

Removed the `monitoring_status` category and associated terms. Expanded terms associated with the `monitoring_frequency` category.

jacob-a-brown commented on db/group.py:
```diff
@@ -31,11 +31,15 @@
+    group_type: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
+    monitoring_frequency: Mapped[Optional[str]] = mapped_column(
+        String(50), nullable=True
+    )
```
these can be lexicon terms

ksmuczynski replied:
Good catch, thank you. The lexicon file should already be up to date.

ksmuczynski committed:
refactor: update `Group` model

`group_type` and `monitoring_frequency` fields should be lexicon terms.

Updated `group_type` and `monitoring_frequency` fields to map to lexicon terms.
